app/models/job_invocation_composer.rb

```diff
@@ -1,49 +1,186 @@
+    end
+
+    # parses params to get job templates in form of id => attributes for selected job templates, e.g.
+    # {
+    #   "459" => {},
+    #   "454" => {
+    #     "input_values" => {
+    #       "2" => {
+    #         "value" => ""
+    #       },
+    #       "5" => {
+    #         "value" => ""
+    #       }
+    #     }
+    #   }
+    # }
+    def template_invocations_params
+      providers_base.values.map do |template_params|
+        template_base = (template_params.fetch(:job_templates, {}).fetch(template_params[:job_template_id], {})).dup.with_indifferent_access
+        template_base[:template_id] = template_params[:job_template_id]
+        input_values_params = template_base.fetch(:input_values, {})
+        template_base[:input_values] = input_values_params.map do |id, values|
+          values.merge(:template_input_id => id)
+        end
+
+        template_base
+      end
+    end
   end
 
-  def compose_from_params(params)
-    @params = params
+  class ApiParams
+    attr_reader :api_params
 
-    @host_ids = validate_host_ids(params[:host_ids])
-    @search_query = targeting_base[:search_query]
+    def initialize(api_params)
+      @api_params = api_params
+    end
 
-    @triggering = ::ForemanTasks::Triggering.new_from_params triggering_base
+    def params
+      { :job_name => job_name,
+        :targeting => targeting_params,
+        :triggering => {},
+        :template_invocations => template_invocations_params }.with_indifferent_access
+    end
 
-    job_invocation.job_name = validate_job_name(job_invocation_base[:job_name])
-    job_invocation.job_name ||= available_job_names.first if job_invocation.new_record?
-    job_invocation.targeting = build_targeting
-    job_invocation.task_group = JobInvocationTaskGroup.new
-    job_invocation.triggering = @triggering
+    def job_name
+      api_params[:job_name]
+    end
 
-    @job_template_ids = validate_job_template_ids(job_templates_base.keys.compact)
-    self
+    def targeting_params
+      raise ::Foreman::Exception, _('Cannot specify both bookmark_id and search_query') if api_params[:bookmark_id] && api_params[:search_query]
+      api_params.slice(:targeting_type, :bookmark_id, :search_query).merge(:user_id => User.current.id)
+    end
+
+    def template_invocations_params
+      template = template_with_default
+      template_invocation_params = { :template_id => template.id, :effective_user => api_params[:effective_user] }
+      template_invocation_params[:input_values] = api_params.fetch(:inputs, []).map do |input_params|
+        input = template.template_inputs.find_by_name(input_params[:name])
+        unless input
+          raise ::Foreman::Exception, _('Unknown input %{input_name} for template %{template_name}') %
+              { :input_name => input_params[:name], :template_name => template.name }
+        end
+        { :template_input_id => input.id, :value => input_params[:value] }
```

Comment:
I've rebased with refelcting the changes in API (both `s/template_id/job_template_id/` and `inputs` format)